src/view.c

```diff
@@ -1231,7 +1288,7 @@ view_snap_to_edge(struct view *view, enum view_edge edge, bool store_natural_geo
 		return;
 	}
 
-	if (view->tiled == edge && !view->maximized) {
+	if (view->tiled == edge && view->maximized == VIEW_AXIS_NONE) {
```

Comment:
Personally I'd prefer keeping the `!view->maximized`, makes it easier to read IMHO. But that is just a personal preference and we could keep it like what you have done here.

Reply:
I think I would prefer to keep the comparison with NONE or BOTH explicit. To me the implicit bool cast is a bit less clear, since the half-maximized states are treated as "maximized" in some cases but "not maximized" in order.

Reply:
> To me the implicit bool cast is a bit less clear, since the half-maximized states are treated as "maximized" in some cases but "not maximized" in others.

Understood. What about `!view->maximized`?

Reply:
I personally don't like using a Boolean operator on an enum value. The equality check is clearer to me.

Reply:
Alright, guess we have different preferences here which is totally fine :)